### PR TITLE
Fix/cross device event indicators 1008 1011

### PR DESCRIPTION
**Pull request by vyeruva-onfido**
# Problem

Fixes for the following bugs:

https://github.com/onfido/onfido-sdk-ui/issues/1008
https://github.com/onfido/onfido-sdk-ui/issues/1011

# Solution

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated? - n/a
- [x] Has the README been updated?
- [ ] Has the RELEASE_GUIDELINES been updated? - n/a
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes? - n/a
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes? - n/a
- [ ] Have any new automated tests been implemented or the existing ones changed? - n/a
- [ ] Have any new manual tests been written down or the existing ones changed? -n/a
- [ ] Have any new strings been translated or the existing ones changed? - n/a

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

File: src/Tracker/index.js [hunk 1]
```diff
@@ -97,7 +98,7 @@ const formatProperties = properties => {
 }
 
 const userAnalyticsEvent = (eventName, properties) => {
-  dispatchEvent(new CustomEvent('userAnalyticsEvent', {detail: {eventName, properties}}));
+  dispatchEvent(new CustomEvent('userAnalyticsEvent', {detail: {eventName, properties, isCrossDevice: false}}));
```
Comment (vyeruva-onfido) [inline src/Tracker/index.js:101]: Is there a way to access the value of globals.mobileConnected(stored in redux)  in this function cleanly? If there is a way to do this, the boolean isCrossDevice can be set to the value of globals.mobileConnected and there is no need to override in Router/index.js

File: src/components/Router/index.js [hunk 2]
```diff
@@ -51,8 +51,9 @@ class CrossDeviceMobileRouter extends Component {
     this.state.socket.open()
     this.requestMobileConfig()
     if (this.props.options.mobileFlow) {
+      this.sendMessage('cross device start', {});
       addEventListener('userAnalyticsEvent', (event) => {
-        this.sendMessage('user analytics', { detail: event.detail } )
+        this.sendMessage('user analytics', { detail: { ... event.detail, isCrossDevice: true} } )
```
Comment (vyeruva-onfido) [inline src/components/Router/index.js:56]: overriding the value here.

File: src/components/crossDevice/MobileFlow.js [hunk 3]
```diff
@@ -43,8 +45,23 @@ class MobileFlow extends Component {
...
+      detail: {
+        eventName: 'CROSS_DEVICE_START',
+        isCrossDevice: true
+      }
+    }));
+  }
+
   onUserAnalyticsEvent = (data) => {
-    dispatchEvent(new CustomEvent('userAnalyticsEvent', data));
+    dispatchEvent(new CustomEvent('userAnalyticsEvent',
```
Comment (vyeruva-onfido) [inline src/components/crossDevice/MobileFlow.js:59]: @vyeruva-onfido revert this change

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

File: src/Tracker/index.js (hunk 1 above)
Comment (stefaniacardenas) [reply src/Tracker/index.js:101]: You cannot easily access the redux store here. I can see that this function is called inside the `sendEvent` which is used inside the `StepsRouter` component. Inside the `StepsRouter` you could access the `mobileFlow` parameter (which tells you if the SDK is used as cross device client or not) from the store or from one of the `StepsRouter` prop. I'm not sure if that will do the trick though because sendEvent is called in different places within the code.

File: src/Tracker/index.js [hunk 4]
```diff
@@ -26,6 +26,7 @@ const integratorTrackedEvents = new Map([
     ['screen_face_video_capture_step_1', 'VIDEO_FACIAL_CAPTURE_STEP_1'],
     ['screen_face_video_capture_step_2', 'VIDEO_FACIAL_CAPTURE_STEP_2'],
     ['Starting upload', 'UPLOAD'],
+    ['screen_document_type_select', 'DOCUMENT_TYPE_SELECT']
```
[the remark this answers is not in the document]
Comment (vyeruva-onfido) [reply src/Tracker/index.js:29]: Will do.

File: src/Tracker/index.js (hunk 1 above)
Comment (vyeruva-onfido) [reply src/Tracker/index.js:101]: @stefaniacardenas Thanks for sharing your thoughts. That was my concern too. I want the code changes to be very less intrusive. So, didn't take that approach.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,8 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 ### Added
 - Internal: Added basic history to SDK demo.
 - Public: Added French translation. The language tag is `fr_FR`.
+- Public: Added `isCrossDevice` flag to user analytics events to differentiate between cross-device and non-cross-device events
+- Public: Added `DOCUMENT_TYPE_SELECT` and `FACIAL_CAPTURE` to user analytics event list
```
Comment (stefaniacardenas) [inline CHANGELOG.md:14]: This should be moved under the next-version header

Comment (JakeBat) [reply CHANGELOG.md:12]: Updated

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Photo/SelfieIntro.js [hunk 6]
```diff
@@ -41,7 +41,10 @@ class Intro extends Component<Props, State> {
       <div className={theme.thickWrapper}>
         <Button
           variants={['primary', 'centered', 'lg']}
-          onClick={nextStep}
+          onClick={() => {
+            sendScreen(['face_selfie_capture'])
```
Comment (stefaniacardenas) [inline src/components/Photo/SelfieIntro.js:45]: What's the full event name that woopra will receive? I just want to make sure we are not sending a duplicate event name

Comment (JakeBat) [reply src/components/Photo/SelfieIntro.js:45]: The full name will be `screen_face_selfie_capture`. I added this event because I did not see it was already present for either woopra or user analytics, please let me know if that's incorrect.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Photo/SelfieIntro.js (hunk 6 above)
Comment (stefaniacardenas) [reply src/components/Photo/SelfieIntro.js:45]: These are the events we are currently present in woopra:
Screenshot 2020-07-09 at 18 15 41.
I think `screen_face_selfie_capture_camera` is sent when the user access the camera, while `screen_face_selfie_capture_file_upload` is sent when the user sees the uploader screen. I think it's fine to send the 'face_selfie_capture'. I will approve

Comment (JakeBat) [reply src/components/Photo/SelfieIntro.js:45]: Thanks @stefaniacardenas, I wasn't able to find `screen_face_selfie_capture_camera` in my local testing. Please let me know if you want me to update.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Photo/SelfieIntro.js (hunk 6 above)
Comment (stefaniacardenas) [reply src/components/Photo/SelfieIntro.js:45]: I think it's ok to just send `screen_face_selfie_capture`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1025-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1025-pr-onfido-sdk-ui-onfido.surge.sh
